File: src/commands/developer/ping.ts
import { TofuCommand } from "@/core/command";
import { emojis } from "@/utils/emojis";
import { ErisUtils } from "@/utils/eris";

export const pingCommand: TofuCommand = {
    config: {
        name: "ping",
        description: "Ping pong!",
    },
    invoke: async () => ({
        message: ErisUtils.prettyMessage(emojis.pingPong, "Pong!"),
    }),
};

File: commands/ping.go
package commands

import (
	"github.com/disgoorg/disgo/discord"
	"github.com/disgoorg/disgo/events"
	"me.zyrouge.tofu/core"
	"me.zyrouge.tofu/utils"
)

func NewTofuPingCommand() core.TofuCommand {
	return core.TofuCommand{
		Config: discord.SlashCommandCreate{
			Name:        "ping",
			Description: "Ping, pong!",
			Contexts: []discord.InteractionContextType{
				discord.InteractionContextTypeGuild,
			},
		},
		Invoke: func(tofu *core.Tofu, event *events.ApplicationCommandInteractionCreate) discord.InteractionResponseData {
			return discord.NewMessageCreateBuilder().
				SetContent(utils.PrettyMessage(utils.EmojiPingPong, "Pong!")).
				Build()
		},
	}
}
